### PR TITLE
modify churn pool selection and add function to do replacement on attributtable clients

### PR DESCRIPTION
**Pull request by irrationalagent**
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is
  referenced, the pull request should include the bug number in the title)
- [ ] Scan the PR and verify that no changes (particularly to
  `.circleci/config.yml`) will cause environment variables (particularly
  credentials) to be exposed in test logs
- [ ] Ensure the container image will be using permissions granted to
  [telemetry-airflow](https://github.com/mozilla/telemetry-airflow/)
  responsibly.

### CODE REVIEW

File: jobs/client-regeneration/client_regeneration/main.py [hunk 1]
```diff
@@ -302,6 +302,109 @@ def update_churn_pool(client, seed, date):
...
+  )
+
+  SELECT
+      * EXCEPT(regened_last_date, rn)
+  FROM numbered
+  WHERE rn = 1
+  """
+
+def write_usage_history(client, seed, start_date, end_date):
+  table_name = """mozdata.analysis.regen_sim_replaced_clients_last_seen_{}""".format(str(seed))
```
Comment (ANich) [inline jobs/client-regeneration/client_regeneration/main.py:267]: Do we need a last seen table? IIRC we want a `clients_yearly` type table. @fbertsch was going to update the initialization of `clients_yearly` to be able to create this in a single query

File: jobs/client-regeneration/client_regeneration/main.py [hunk 2]
```diff
@@ -302,6 +302,109 @@ def update_churn_pool(client, seed, date):
...
+      SELECT
+        COALESCE(r.replacement_id, c.client_id) AS client_id,
+        COALESCE(r.first_seen_date, c.first_seen_date) AS first_seen_date,
+        c.client_id AS original_client_id,
+        c.first_seen_date AS original_first_seen_date,
+        c.submission_date,
+        c.country,
+        c.device_model,
+        regened_last_date,
+
```
Comment (ANich) [inline jobs/client-regeneration/client_regeneration/main.py:243]: Do we need a `sample_id` as well? I believe that was used for train/test split in the notebook

**Review by ANich — COMMENTED** [review]
Left a few questions

File: jobs/client-regeneration/client_regeneration/main.py (hunk 1 above)
Comment (irrationalagent) [reply jobs/client-regeneration/client_regeneration/main.py:267]: I guess its mainly a stopgap. I've been using it to analyze / validate retention on the replaced data. It makes comparison to the existing clients_last_seen easy

**Review by ANich — APPROVED** [review]

**Pull request merged**